core/shared/src/main/scala/org/http4s/CharsetRange.scala

```diff
@@ -44,13 +44,19 @@ object CharsetRange {
 
   object `*` extends `*`(QValue.One)
 
-  final case class Atom protected[http4s] (charset: Charset, qValue: QValue = QValue.One)
-      extends CharsetRange {
+  final case class Atom private[http4s] ( // scalafix:ok Http4sGeneralLinters.nonValidatingCopyConstructor; bincompat until 1.0
+      charset: Charset,
+      qValue: QValue = QValue.One,
+  ) extends CharsetRange {
     override def withQValue(q: QValue): CharsetRange.Atom = copy(qValue = q)
 
     def render(writer: Writer): writer.type = writer << charset << qValue
   }
 
+  object Atom extends scala.runtime.AbstractFunction2[Charset, QValue, Atom] {
```

Comment:
This extension seems bizarre 🤔

Reply:
Thanks to @som-snytt for pointing me on the right track. It's not directly related to `-Xsource:3`.

In Scala 2, if the companion `object` is synthesized then it extends `AbstractFunctionN`. But if the `object` is explicitly in the source (as it is here, now) then it does not extend `AbstractFunctionN` automatically, so we need to do so manually.